### ISSUE TITLE
spelled intervals don't add correctly

### ISSUE THREAD

**Issue by robert-lieck** [DCMLab/pitchtypes#7]
Octave "overflow" is not handled correctly:
```python
for i in ["+m2:0", "+P5:0"]:
    i = SpelledInterval(i)
    print(i, i.value)
    print(i + i, (i + i).value)
```
produces
```python
+m2:0 [-5  0]
+d3:0 [-10   0]
+P5:0 [1 0]
+M2:0 [2 0]
```
The first case is correct (I think). But in the second case, the fifths and octaves are separately added and the result is off by one octave; it should be `+M2:1`.

**Comment by chfin** [comment]
I think the right way to check this is to look at the diatonic steps encoded by the fifths, add them, and count the number of octaves (basically downward rounding `octs = steps / 7`). The diatonic steps can be obtained using `steps = fifths * 4 % 7` (where `% 7` turns specific into generic intervals).

A similar issue exists with inversion, i.e. `M2:0 = [2,0]` but `-M2:0 = [-2,-1] = m7:-1` (since `[-2,0] = m7:0`). **But** you can't just correct the octaves by `-1` on inversion since this doesn't hold for unisons: `a1 = [7,0]`, `-a1 = d1 = [-7,0]`.

**Comment by robert-lieck** [comment]
We should also check for correct inversion in the unittests (#11).

**Comment by robert-lieck** [comment]
This was fixed, mainly in 47e09ee892cad0435c1e1b77c3fec8040f5ba6c8, 72caf1d4a12c099015abca999b5c1fd66240561f, and fdc9fa99e7b617e4fa4de0eba07060c2d9ad11c8.
